e107_handlers/cache_handler.php

```diff
@@ -146,13 +144,13 @@ public function retrieve($CacheTag, $MaximumAge = false, $ForcedCheck = false, $
 				else
 				{
 					$ret = file_get_contents($cache_file);
-					if (substr($ret,0,strlen(self::CACHE_PREFIX)) == self::CACHE_PREFIX)
+					if (strpos($ret,self::CACHE_PREFIX) === 0 )  
 					{
 						$ret = substr($ret, strlen(self::CACHE_PREFIX));
 					}
-					elseif(substr($ret,0,5) == '<?php')
+					elseif(strpos($ret,'<?php') === 0)
 					{
-						$ret = substr($ret, 5);		// Handle the history for now
+						$ret = '<?php' ;			// Handle the history for now
```

Comment:
Maybe something went wrong. It doesn't look fixed here: 

					elseif(strpos($ret,'<?php') === 0)
					{
						$ret = '<?php' ;			// Handle the history for now
					}